cogip/tools/mcu_logger/__main__.py

```diff
@@ -35,6 +36,8 @@ def main_opt(
             if ser.in_waiting > 0:
                 line = ser.readline().decode("utf-8").strip()
                 logger.info(line)
+            else:
+                sleep(0.01)
```

Comment:
What is consuming here is the waking up of the CPU in an infinite loop to check `ser.in_waiting`.

You do not need all of this as readline() is already blocking as you did not defined any timeout (default to None).

I would write:
```python
try:
    while True:
        line = ser.readline().decode("utf-8").strip()
        logger.info(line)
except KeyboardInterrupt:
    pass
```
If you absolutely want a timeout, define it inside `ser = serial.Serial(str(port), baud_rate, timeout=0.1)`

Reply:
Done but not tested.